File: scala-v1-basic/producer/src/main/scala/clients/BasicProducer.scala
package clients

import java.time.Duration
import java.util.Properties

import org.apache.kafka.clients.producer.ProducerConfig.{BOOTSTRAP_SERVERS_CONFIG, CLIENT_ID_CONFIG, KEY_SERIALIZER_CLASS_CONFIG, VALUE_SERIALIZER_CLASS_CONFIG}
import org.apache.kafka.clients.producer.{KafkaProducer, ProducerRecord}
import org.apache.kafka.common.serialization.StringSerializer

object BasicProducer {

  def main(args: Array[String]): Unit = {

    println("*** Starting Basic Producer ***")

    val settings = new Properties()

    settings.put(CLIENT_ID_CONFIG, "basic-producer")
    settings.put(BOOTSTRAP_SERVERS_CONFIG, "kafka:9092")
    settings.put(KEY_SERIALIZER_CLASS_CONFIG, classOf[StringSerializer].getCanonicalName)
    settings.put(VALUE_SERIALIZER_CLASS_CONFIG, classOf[StringSerializer].getCanonicalName)

    val producer = new KafkaProducer[String, String](settings)

    val topic = "basic-topic"

    for (i <- 1 to 5) {
      val key = "key-" + i
      val value = "value-" + i

      val record = new ProducerRecord[String, String](topic, key, value)

      producer.send(record)
    }

    producer.close(Duration.ofMillis(100))

    println("### Stopping Basic Producer ###")

  }
}

File: java-starter/producer/src/main/java/net/chrissearle/kafka/BasicProducer.java
package net.chrissearle.kafka;

import java.util.Properties;

import org.apache.kafka.clients.producer.KafkaProducer;
import org.apache.kafka.clients.producer.ProducerRecord;

public class BasicProducer {

  public static final String TOPIC = "hello-world-topic";

  public static void main(String[] args) {

    System.out.println("*** Starting Basic Producer ***");

    Properties settings = new Properties();

    settings.put("client.id", "basic-producer");
    settings.put("bootstrap.servers", "localhost:29092");
    settings.put("key.serializer", "org.apache.kafka.common.serialization.StringSerializer");
    settings.put("value.serializer", "org.apache.kafka.common.serialization.StringSerializer");

    try (KafkaProducer<String, String> producer = new KafkaProducer<>(settings)) {
      for (int i = 1; i <= 5; i++) {
        final String key = "key-" + i;
        final String value = "value-" + i;

        System.out.println("### Sending " + i + " ###");

        producer.send(new ProducerRecord<>(TOPIC, key, value));
      }
    }
  }
}
